detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -38,27 +41,65 @@ import { getEnv } from '@opentelemetry/core';
 export class AwsEcsDetector implements Detector {
   readonly CONTAINER_ID_LENGTH = 64;
   readonly DEFAULT_CGROUP_PATH = '/proc/self/cgroup';
+
   private static readFileAsync = util.promisify(fs.readFile);
 
+  private static async getUrlAsJson(url: string): Promise<any> {
+    return new Promise<string>((resolve, reject) => {
+      const request = http.get(url, (response: any) => {
+        if (response.statusCode >= 400) {
+          reject(`Request to '${url}' failed with status ${response.statusCode}`);
```

Comment:
It is more typical to reject with an `Error` instead of a string since the catch clause often expects an error.

Reply:
Done in 5a15762f, also added a test that was missing